apis/storage/v1alpha3/policy_types.go

```diff
@@ -0,0 +1,63 @@
+package v1alpha3
+
+import "encoding/json"
+
+// S3BucketPolicy defines the policy for the S3 Bucket being created.
+type S3BucketPolicy struct {
+	// This is the current IAM policy version
+	PolicyVersion string `json:"Version"`
+
+	// This is the policy's optional identifier
+	PolicyID string `json:"Id,omitempty"`
+
+	PolicyStatement []S3BucketPolicyStatement `json:"Statement"`
```

Comment:
Are you thinking the `PolicyStatement` should be structured for the user instead of inline JSON? I could see advantages to that because we could have something like a `PrincipalRef`, but we also could allow the user to do some sort of templating where they could say they wanted a JSON value populated by a `BucketRef`. The former is probably a little bit simpler of a user experience, but also means that a user would not be able to copy a JSON policy and use it directly.

Reply:
Would you suggest I add an additional field to allow the user to directly supply the policy, then when that one is supplied it will take precedence over any generation?